File: binding/include/opengl/libqopengldebug.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qopengldebugmessage.html
pub const qopengldebugmessage = struct {

    /// New constructs a new QOpenGLDebugMessage object.
    pub fn New() QtC.QOpenGLDebugMessage {
        return qtc.QOpenGLDebugMessage_new();
    }


    /// New2 constructs a new QOpenGLDebugMessage object.
    pub fn New2(debugMessage: ?*anyopaque) QtC.QOpenGLDebugMessage {
        return qtc.QOpenGLDebugMessage_new2(@ptrCast(debugMessage));
    }


    pub fn OperatorAssign(self: ?*anyopaque, debugMessage: ?*anyopaque) void {
        qtc.QOpenGLDebugMessage_OperatorAssign(@ptrCast(self), @ptrCast(debugMessage));
    }

    pub fn Swap(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QOpenGLDebugMessage_Swap(@ptrCast(self), @ptrCast(other));
    }

    pub fn Source(self: ?*anyopaque, ) i64 {
        return qtc.QOpenGLDebugMessage_Source(@ptrCast(self));
    }

    pub fn Type(self: ?*anyopaque, ) i32 {
        return qtc.QOpenGLDebugMessage_Type(@ptrCast(self));
    }

    pub fn Severity(self: ?*anyopaque, ) i64 {
        return qtc.QOpenGLDebugMessage_Severity(@ptrCast(self));
    }

    pub fn Id(self: ?*anyopaque, ) u32 {
        return qtc.QOpenGLDebugMessage_Id(@ptrCast(self));
    }

    pub fn Message(self: ?*anyopaque, , allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QOpenGLDebugMessage_Message(@ptrCast(self));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qopengldebugmessage.Message: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn CreateApplicationMessage(text: []const u8) QtC.QOpenGLDebugMessage {
        const text_str = qtc.libqt_string{
    .len = text.len,
    .data = text.ptr,
};
return qtc.QOpenGLDebugMessage_CreateApplicationMessage(text_str);
    }

    pub fn CreateThirdPartyMessage(text: []const u8) QtC.QOpenGLDebugMessage {
        const text_str = qtc.libqt_string{
    .len = text.len,
    .data = text.ptr,
};
return qtc.QOpenGLDebugMessage_CreateThirdPartyMessage(text_str);
    }

    pub fn OperatorEqual(self: ?*anyopaque, debugMessage: ?*anyopaque) bool {
        return qtc.QOpenGLDebugMessage_OperatorEqual(@ptrCast(self), @ptrCast(debugMessage));
    }

    pub fn OperatorNotEqual(self: ?*anyopaque, debugMessage: ?*anyopaque) bool {
        return qtc.QOpenGLDebugMessage_OperatorNotEqual(@ptrCast(self), @ptrCast(debugMessage));
    }

    pub fn CreateApplicationMessage2(text: []const u8, id: u32) QtC.QOpenGLDebugMessage {
        const text_str = qtc.libqt_string{
    .len = text.len,
    .data = text.ptr,
};
return qtc.QOpenGLDebugMessage_CreateApplicationMessage2(text_str, id);
    }

    pub fn CreateApplicationMessage3(text: []const u8, id: u32, severity: i64) QtC.QOpenGLDebugMessage {
        const text_str = qtc.libqt_string{
    .len = text.len,
    .data = text.ptr,
};
return qtc.QOpenGLDebugMessage_CreateApplicationMessage3(text_str, id, @intCast(severity));
    }

    pub fn CreateApplicationMessage4(text: []const u8, id: u32, severity: i64, typeVal: i32) QtC.QOpenGLDebugMessage {
        const text_str = qtc.libqt_string{
    .len = text.len,
    .data = text.ptr,
};
return qtc.QOpenGLDebugMessage_CreateApplicationMessage4(text_str, id, @intCast(severity), @intCast(typeVal));
    }

    pub fn CreateThirdPartyMessage2(text: []const u8, id: u32) QtC.QOpenGLDebugMessage {
        const text_str = qtc.libqt_string{
    .len = text.len,
    .data = text.ptr,
};
return qtc.QOpenGLDebugMessage_CreateThirdPartyMessage2(text_str, id);
    }

    pub fn CreateThirdPartyMessage3(text: []const u8, id: u32, severity: i64) QtC.QOpenGLDebugMessage {
        const text_str = qtc.libqt_string{
    .len = text.len,
    .data = text.ptr,
};
return qtc.QOpenGLDebugMessage_CreateThirdPartyMessage3(text_str, id, @intCast(severity));
    }

    pub fn CreateThirdPartyMessage4(text: []const u8, id: u32, severity: i64, typeVal: i32) QtC.QOpenGLDebugMessage {
        const text_str = qtc.libqt_string{
    .len = text.len,
    .data = text.ptr,
};
return qtc.QOpenGLDebugMessage_CreateThirdPartyMessage4(text_str, id, @intCast(severity), @intCast(typeVal));
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QOpenGLDebugMessage_Delete(@ptrCast(self));
    }
};

/// https://doc.qt.io/qt-6/qopengldebuglogger.html
pub const qopengldebuglogger = struct {

    /// New constructs a new QOpenGLDebugLogger object.
    pub fn New() QtC.QOpenGLDebugLogger {
        return qtc.QOpenGLDebugLogger_new();
    }


    /// New2 constructs a new QOpenGLDebugLogger object.
    pub fn New2(parent: ?*anyopaque) QtC.QOpenGLDebugLogger {
        return qtc.QOpenGLDebugLogger_new2(@ptrCast(parent));
    }


    pub fn Tr(s: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QOpenGLDebugLogger_Tr(@ptrCast(s));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qopengldebuglogger.Tr: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Initialize(self: ?*anyopaque, ) bool {
        return qtc.QOpenGLDebugLogger_Initialize(@ptrCast(self));
    }

    pub fn IsLogging(self: ?*anyopaque, ) bool {
        return qtc.QOpenGLDebugLogger_IsLogging(@ptrCast(self));
    }

    pub fn LoggingMode(self: ?*anyopaque, ) i32 {
        return qtc.QOpenGLDebugLogger_LoggingMode(@ptrCast(self));
    }

    pub fn MaximumMessageLength(self: ?*anyopaque, ) i64 {
        return qtc.QOpenGLDebugLogger_MaximumMessageLength(@ptrCast(self));
    }

    pub fn PushGroup(self: ?*anyopaque, name: []const u8) void {
        const name_str = qtc.libqt_string{
    .len = name.len,
    .data = name.ptr,
};
qtc.QOpenGLDebugLogger_PushGroup(@ptrCast(self), name_str);
    }

    pub fn PopGroup(self: ?*anyopaque, ) void {
        qtc.QOpenGLDebugLogger_PopGroup(@ptrCast(self));
    }

    pub fn EnableMessages(self: ?*anyopaque, ) void {
        qtc.QOpenGLDebugLogger_EnableMessages(@ptrCast(self));
    }

    pub fn EnableMessages2(self: ?*anyopaque, ids: []const u8) void {
        qtc.QOpenGLDebugLogger_EnableMessages2(@ptrCast(self), @ptrCast(ids));
    }

    pub fn DisableMessages(self: ?*anyopaque, ) void {
        qtc.QOpenGLDebugLogger_DisableMessages(@ptrCast(self));
    }

    pub fn DisableMessages2(self: ?*anyopaque, ids: []const u8) void {
        qtc.QOpenGLDebugLogger_DisableMessages2(@ptrCast(self), @ptrCast(ids));
    }

    pub fn LoggedMessages(self: ?*anyopaque, ) []const u8 {
        return qtc.QOpenGLDebugLogger_LoggedMessages(@ptrCast(self));
    }

    pub fn LogMessage(self: ?*anyopaque, debugMessage: ?*anyopaque) void {
        qtc.QOpenGLDebugLogger_LogMessage(@ptrCast(self), @ptrCast(debugMessage));
    }

    pub fn StartLogging(self: ?*anyopaque, ) void {
        qtc.QOpenGLDebugLogger_StartLogging(@ptrCast(self));
    }

    pub fn StopLogging(self: ?*anyopaque, ) void {
        qtc.QOpenGLDebugLogger_StopLogging(@ptrCast(self));
    }

    pub fn MessageLogged(self: ?*anyopaque, debugMessage: ?*anyopaque) void {
        qtc.QOpenGLDebugLogger_MessageLogged(@ptrCast(self), @ptrCast(debugMessage));
    }

    pub fn OnMessageLogged(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QOpenGLDebugLogger_Connect_MessageLogged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Tr2(s: []const u8, c: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QOpenGLDebugLogger_Tr2(@ptrCast(s), @ptrCast(c));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qopengldebuglogger.Tr2: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Tr3(s: []const u8, c: []const u8, n: i32, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QOpenGLDebugLogger_Tr3(@ptrCast(s), @ptrCast(c), n);
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qopengldebuglogger.Tr3: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn PushGroup2(self: ?*anyopaque, name: []const u8, id: u32) void {
        const name_str = qtc.libqt_string{
    .len = name.len,
    .data = name.ptr,
};
qtc.QOpenGLDebugLogger_PushGroup2(@ptrCast(self), name_str, id);
    }

    pub fn PushGroup3(self: ?*anyopaque, name: []const u8, id: u32, source: i64) void {
        const name_str = qtc.libqt_string{
    .len = name.len,
    .data = name.ptr,
};
qtc.QOpenGLDebugLogger_PushGroup3(@ptrCast(self), name_str, id, @intCast(source));
    }

    pub fn EnableMessages1(self: ?*anyopaque, sources: i64) void {
        qtc.QOpenGLDebugLogger_EnableMessages1(@ptrCast(self), @intCast(sources));
    }

    pub fn EnableMessages22(self: ?*anyopaque, sources: i64, types: i64) void {
        qtc.QOpenGLDebugLogger_EnableMessages22(@ptrCast(self), @intCast(sources), @intCast(types));
    }

    pub fn EnableMessages3(self: ?*anyopaque, sources: i64, types: i64, severities: i64) void {
        qtc.QOpenGLDebugLogger_EnableMessages3(@ptrCast(self), @intCast(sources), @intCast(types), @intCast(severities));
    }

    pub fn EnableMessages23(self: ?*anyopaque, ids: []const u8, sources: i64) void {
        qtc.QOpenGLDebugLogger_EnableMessages23(@ptrCast(self), @ptrCast(ids), @intCast(sources));
    }

    pub fn EnableMessages32(self: ?*anyopaque, ids: []const u8, sources: i64, types: i64) void {
        qtc.QOpenGLDebugLogger_EnableMessages32(@ptrCast(self), @ptrCast(ids), @intCast(sources), @intCast(types));
    }

    pub fn DisableMessages1(self: ?*anyopaque, sources: i64) void {
        qtc.QOpenGLDebugLogger_DisableMessages1(@ptrCast(self), @intCast(sources));
    }

    pub fn DisableMessages22(self: ?*anyopaque, sources: i64, types: i64) void {
        qtc.QOpenGLDebugLogger_DisableMessages22(@ptrCast(self), @intCast(sources), @intCast(types));
    }

    pub fn DisableMessages3(self: ?*anyopaque, sources: i64, types: i64, severities: i64) void {
        qtc.QOpenGLDebugLogger_DisableMessages3(@ptrCast(self), @intCast(sources), @intCast(types), @intCast(severities));
    }

    pub fn DisableMessages23(self: ?*anyopaque, ids: []const u8, sources: i64) void {
        qtc.QOpenGLDebugLogger_DisableMessages23(@ptrCast(self), @ptrCast(ids), @intCast(sources));
    }

    pub fn DisableMessages32(self: ?*anyopaque, ids: []const u8, sources: i64, types: i64) void {
        qtc.QOpenGLDebugLogger_DisableMessages32(@ptrCast(self), @ptrCast(ids), @intCast(sources), @intCast(types));
    }

    pub fn StartLogging1(self: ?*anyopaque, loggingMode: i32) void {
        qtc.QOpenGLDebugLogger_StartLogging1(@ptrCast(self), @intCast(loggingMode));
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QOpenGLDebugLogger_Delete(@ptrCast(self));
    }
};

/// https://doc.qt.io/qt-6/opengldebug.html#types
pub const enums = struct {
    pub const Source = enum {
        pub const InvalidSource: i64 = 0;
        pub const APISource: i64 = 1;
        pub const WindowSystemSource: i64 = 2;
        pub const ShaderCompilerSource: i64 = 4;
        pub const ThirdPartySource: i64 = 8;
        pub const ApplicationSource: i64 = 16;
        pub const OtherSource: i64 = 32;
        pub const LastSource: i64 = 32;
        pub const AnySource: i64 = 4294967295;
    };

    pub const Type = enum {
        pub const InvalidType: i64 = 0;
        pub const ErrorType: i64 = 1;
        pub const DeprecatedBehaviorType: i64 = 2;
        pub const UndefinedBehaviorType: i64 = 4;
        pub const PortabilityType: i64 = 8;
        pub const PerformanceType: i64 = 16;
        pub const OtherType: i64 = 32;
        pub const MarkerType: i64 = 64;
        pub const GroupPushType: i64 = 128;
        pub const GroupPopType: i64 = 256;
        pub const LastType: i64 = 256;
        pub const AnyType: i64 = 4294967295;
    };

    pub const Severity = enum {
        pub const InvalidSeverity: i64 = 0;
        pub const HighSeverity: i64 = 1;
        pub const MediumSeverity: i64 = 2;
        pub const LowSeverity: i64 = 4;
        pub const NotificationSeverity: i64 = 8;
        pub const LastSeverity: i64 = 8;
        pub const AnySeverity: i64 = 4294967295;
    };

    pub const LoggingMode = enum {
        pub const AsynchronousLogging: i32 = 0;
        pub const SynchronousLogging: i32 = 1;
    };

};

File: binding/src/opengl/libqopengldebug.cpp
#include <QObject>
#include <QOpenGLDebugLogger>
#include <QOpenGLDebugMessage>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <qopengldebug.h>
#include "libqopengldebug.h"
#include "libqopengldebug.hxx"

QOpenGLDebugMessage* QOpenGLDebugMessage_new() {
	 return new QOpenGLDebugMessage();
}

QOpenGLDebugMessage* QOpenGLDebugMessage_new2(const QOpenGLDebugMessage* debugMessage) {
	 return new QOpenGLDebugMessage(*debugMessage);
}

void QOpenGLDebugMessage_OperatorAssign(QOpenGLDebugMessage* self, const QOpenGLDebugMessage* debugMessage) {
	self->operator=(*debugMessage);
}

void QOpenGLDebugMessage_Swap(QOpenGLDebugMessage* self, QOpenGLDebugMessage* other) {
	self->swap(*other);
}

int QOpenGLDebugMessage_Source(const QOpenGLDebugMessage* self) {
	return self->source();
}

int QOpenGLDebugMessage_Type(const QOpenGLDebugMessage* self) {
	return self->type();
}

int QOpenGLDebugMessage_Severity(const QOpenGLDebugMessage* self) {
	return self->severity();
}

uint32_t QOpenGLDebugMessage_Id(const QOpenGLDebugMessage* self) {
	return self->id();
}

libqt_string QOpenGLDebugMessage_Message(const QOpenGLDebugMessage* self) {
	QString _ret = self->message();
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

QOpenGLDebugMessage* QOpenGLDebugMessage_CreateApplicationMessage(const libqt_string text) {
	return new QOpenGLDebugMessage(QOpenGLDebugMessage::createApplicationMessage(QString::fromUtf8(text.data, text.len)));
}

QOpenGLDebugMessage* QOpenGLDebugMessage_CreateThirdPartyMessage(const libqt_string text) {
	return new QOpenGLDebugMessage(QOpenGLDebugMessage::createThirdPartyMessage(QString::fromUtf8(text.data, text.len)));
}

bool QOpenGLDebugMessage_OperatorEqual(const QOpenGLDebugMessage* self, const QOpenGLDebugMessage* debugMessage) {
	return self->operator==(*debugMessage);
}

bool QOpenGLDebugMessage_OperatorNotEqual(const QOpenGLDebugMessage* self, const QOpenGLDebugMessage* debugMessage) {
	return self->operator!=(*debugMessage);
}

QOpenGLDebugMessage* QOpenGLDebugMessage_CreateApplicationMessage2(const libqt_string text, uint32_t id) {
	return new QOpenGLDebugMessage(QOpenGLDebugMessage::createApplicationMessage(QString::fromUtf8(text.data, text.len), id));
}

QOpenGLDebugMessage* QOpenGLDebugMessage_CreateApplicationMessage3(const libqt_string text, uint32_t id, int severity) {
	return new QOpenGLDebugMessage(QOpenGLDebugMessage::createApplicationMessage(QString::fromUtf8(text.data, text.len), id, static_cast<QOpenGLDebugMessage::Severity>(severity)));
}

QOpenGLDebugMessage* QOpenGLDebugMessage_CreateApplicationMessage4(const libqt_string text, uint32_t id, int severity, int typeVal) {
	return new QOpenGLDebugMessage(QOpenGLDebugMessage::createApplicationMessage(QString::fromUtf8(text.data, text.len), id, static_cast<QOpenGLDebugMessage::Severity>(severity), static_cast<QMetaType::Type>(typeVal)));
}

QOpenGLDebugMessage* QOpenGLDebugMessage_CreateThirdPartyMessage2(const libqt_string text, uint32_t id) {
	return new QOpenGLDebugMessage(QOpenGLDebugMessage::createThirdPartyMessage(QString::fromUtf8(text.data, text.len), id));
}

QOpenGLDebugMessage* QOpenGLDebugMessage_CreateThirdPartyMessage3(const libqt_string text, uint32_t id, int severity) {
	return new QOpenGLDebugMessage(QOpenGLDebugMessage::createThirdPartyMessage(QString::fromUtf8(text.data, text.len), id, static_cast<QOpenGLDebugMessage::Severity>(severity)));
}

QOpenGLDebugMessage* QOpenGLDebugMessage_CreateThirdPartyMessage4(const libqt_string text, uint32_t id, int severity, int typeVal) {
	return new QOpenGLDebugMessage(QOpenGLDebugMessage::createThirdPartyMessage(QString::fromUtf8(text.data, text.len), id, static_cast<QOpenGLDebugMessage::Severity>(severity), static_cast<QMetaType::Type>(typeVal)));
}

void QOpenGLDebugMessage_Delete(QOpenGLDebugMessage* self) {
    delete self;
}

QOpenGLDebugLogger* QOpenGLDebugLogger_new() {
	 return new QOpenGLDebugLogger();
}

QOpenGLDebugLogger* QOpenGLDebugLogger_new2(QObject* parent) {
	 return new QOpenGLDebugLogger(parent);
}

libqt_string QOpenGLDebugLogger_Tr(const char* s) {
	QString _ret = QOpenGLDebugLogger::tr(s);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

bool QOpenGLDebugLogger_Initialize(QOpenGLDebugLogger* self) {
	return self->initialize();
}

bool QOpenGLDebugLogger_IsLogging(const QOpenGLDebugLogger* self) {
	return self->isLogging();
}

int QOpenGLDebugLogger_LoggingMode(const QOpenGLDebugLogger* self) {
	return self->loggingMode();
}

long long QOpenGLDebugLogger_MaximumMessageLength(const QOpenGLDebugLogger* self) {
	return self->maximumMessageLength();
}

void QOpenGLDebugLogger_PushGroup(QOpenGLDebugLogger* self, const libqt_string name) {
	self->pushGroup(QString::fromUtf8(name.data, name.len));
}

void QOpenGLDebugLogger_PopGroup(QOpenGLDebugLogger* self) {
	self->popGroup();
}

void QOpenGLDebugLogger_EnableMessages(QOpenGLDebugLogger* self) {
	self->enableMessages();
}

void QOpenGLDebugLogger_EnableMessages2(QOpenGLDebugLogger* self, const libqt_list ids) {
	self->enableMessages(*ids);
}

void QOpenGLDebugLogger_DisableMessages(QOpenGLDebugLogger* self) {
	self->disableMessages();
}

void QOpenGLDebugLogger_DisableMessages2(QOpenGLDebugLogger* self, const libqt_list ids) {
	self->disableMessages(*ids);
}

libqt_list QOpenGLDebugLogger_LoggedMessages(const QOpenGLDebugLogger* self) {
	return self->loggedMessages();
}

void QOpenGLDebugLogger_LogMessage(QOpenGLDebugLogger* self, const QOpenGLDebugMessage* debugMessage) {
	self->logMessage(*debugMessage);
}

void QOpenGLDebugLogger_StartLogging(QOpenGLDebugLogger* self) {
	self->startLogging();
}

void QOpenGLDebugLogger_StopLogging(QOpenGLDebugLogger* self) {
	self->stopLogging();
}

void QOpenGLDebugLogger_MessageLogged(QOpenGLDebugLogger* self, const QOpenGLDebugMessage* debugMessage) {
	self->messageLogged(*debugMessage);
}

void QOpenGLDebugLogger_Connect_MessageLogged(QOpenGLDebugLogger* self, intptr_t slot) {
    void (*slotFunc)(QOpenGLDebugLogger*, const QOpenGLDebugMessage*) = reinterpret_cast<void (*)(QOpenGLDebugLogger*, const QOpenGLDebugMessage*)>(slot);
    QOpenGLDebugLogger::connect(self, &QOpenGLDebugLogger::messageLogged, [self, slotFunc](const QOpenGLDebugMessage& debugMessage) {
	slotFunc(self, debugMessage);
    });
}

libqt_string QOpenGLDebugLogger_Tr2(const char* s, const char* c) {
	QString _ret = QOpenGLDebugLogger::tr(s, c);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QOpenGLDebugLogger_Tr3(const char* s, const char* c, int n) {
	QString _ret = QOpenGLDebugLogger::tr(s, c, n);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

void QOpenGLDebugLogger_PushGroup2(QOpenGLDebugLogger* self, const libqt_string name, uint32_t id) {
	self->pushGroup(QString::fromUtf8(name.data, name.len), id);
}

void QOpenGLDebugLogger_PushGroup3(QOpenGLDebugLogger* self, const libqt_string name, uint32_t id, int source) {
	self->pushGroup(QString::fromUtf8(name.data, name.len), id, static_cast<QOpenGLDebugMessage::Source>(source));
}

void QOpenGLDebugLogger_EnableMessages1(QOpenGLDebugLogger* self, int sources) {
	self->enableMessages(static_cast<QFlags<QOpenGLDebugMessage::Source>>(sources));
}

void QOpenGLDebugLogger_EnableMessages22(QOpenGLDebugLogger* self, int sources, int types) {
	self->enableMessages(static_cast<QFlags<QOpenGLDebugMessage::Source>>(sources), static_cast<QFlags<QOpenGLDebugMessage::Type>>(types));
}

void QOpenGLDebugLogger_EnableMessages3(QOpenGLDebugLogger* self, int sources, int types, int severities) {
	self->enableMessages(static_cast<QFlags<QOpenGLDebugMessage::Source>>(sources), static_cast<QFlags<QOpenGLDebugMessage::Type>>(types), static_cast<QFlags<QOpenGLDebugMessage::Severity>>(severities));
}

void QOpenGLDebugLogger_EnableMessages23(QOpenGLDebugLogger* self, const libqt_list ids, int sources) {
	self->enableMessages(*ids, static_cast<QFlags<QOpenGLDebugMessage::Source>>(sources));
}

void QOpenGLDebugLogger_EnableMessages32(QOpenGLDebugLogger* self, const libqt_list ids, int sources, int types) {
	self->enableMessages(*ids, static_cast<QFlags<QOpenGLDebugMessage::Source>>(sources), static_cast<QFlags<QOpenGLDebugMessage::Type>>(types));
}

void QOpenGLDebugLogger_DisableMessages1(QOpenGLDebugLogger* self, int sources) {
	self->disableMessages(static_cast<QFlags<QOpenGLDebugMessage::Source>>(sources));
}

void QOpenGLDebugLogger_DisableMessages22(QOpenGLDebugLogger* self, int sources, int types) {
	self->disableMessages(static_cast<QFlags<QOpenGLDebugMessage::Source>>(sources), static_cast<QFlags<QOpenGLDebugMessage::Type>>(types));
}

void QOpenGLDebugLogger_DisableMessages3(QOpenGLDebugLogger* self, int sources, int types, int severities) {
	self->disableMessages(static_cast<QFlags<QOpenGLDebugMessage::Source>>(sources), static_cast<QFlags<QOpenGLDebugMessage::Type>>(types), static_cast<QFlags<QOpenGLDebugMessage::Severity>>(severities));
}

void QOpenGLDebugLogger_DisableMessages23(QOpenGLDebugLogger* self, const libqt_list ids, int sources) {
	self->disableMessages(*ids, static_cast<QFlags<QOpenGLDebugMessage::Source>>(sources));
}

void QOpenGLDebugLogger_DisableMessages32(QOpenGLDebugLogger* self, const libqt_list ids, int sources, int types) {
	self->disableMessages(*ids, static_cast<QFlags<QOpenGLDebugMessage::Source>>(sources), static_cast<QFlags<QOpenGLDebugMessage::Type>>(types));
}

void QOpenGLDebugLogger_StartLogging1(QOpenGLDebugLogger* self, int loggingMode) {
	self->startLogging(static_cast<QOpenGLDebugLogger::LoggingMode>(loggingMode));
}

void QOpenGLDebugLogger_Delete(QOpenGLDebugLogger* self) {
    delete self;
}

